src/cmps/PokemonTable.tsx

```diff
@@ -28,13 +29,18 @@ import { NO_VALUE } from "../assets/constants/placeholders";
 
 interface PokemonTableProps {
   pokemons: Pokemon[];
+  selectedCtg: number;
 }
 
-const PokemonTable = ({ pokemons }: PokemonTableProps) => {
+const PokemonTable = ({ pokemons, selectedCtg }: PokemonTableProps) => {
   const [page, setPage] = useState(0);
   const [rowsPerPage, setRowsPerPage] = useState(10);
   const [selectedPokemon, setSelectedPokemon] = useState<Pokemon | null>(null);
 
+  useEffect(() => {
+    setPage(0);
+  }, [selectedCtg]);
```

Comment:
every change of ctg you need to reset the page?

Reply:
for this case:
user is in All Pokemons tab, using pagination till page number 10, switch to My Pokemons tab, but user only have 3 pokemons, so i need to set the pagination to page number 0